core/src/components/select/select.tsx

```diff
@@ -310,66 +310,89 @@ export class Select implements ComponentInterface {
-         * when focusing.
+         * If no value is set then focus the first enabled option.
          */
-        const interactiveEl = selectedItem.querySelector<HTMLElement>('ion-radio, ion-checkbox') as
-          | HTMLIonRadioElement
-          | HTMLIonCheckboxElement
-          | null;
-        if (interactiveEl) {
-          // Needs to be called before `focusVisibleElement` to prevent issue with focus event bubbling
-          // and removing `ion-focused` style
-          interactiveEl.setFocus();
+        const firstEnabledOption = overlay.querySelector<HTMLElement>(
+          'ion-radio:not(.radio-disabled), ion-checkbox:not(.checkbox-disabled)'
+        ) as HTMLIonRadioElement | HTMLIonCheckboxElement | null;
+
+        if (firstEnabledOption) {
+          /**
+           * Focus the option for the same reason as we do above.
+           *
+           * Needs to be called before `focusVisibleElement` to prevent issue with focus event bubbling
+           * and removing `ion-focused` style
+           */
+          firstEnabledOption.setFocus();
+
+          focusVisibleElement(firstEnabledOption.closest('ion-item')!);
         }
-
-        focusVisibleElement(selectedItem);
       }
-    } else {
-      /**
-       * If no value is set then focus the first enabled option.
-       */
-      const firstEnabledOption = overlay.querySelector<HTMLElement>(
-        'ion-radio:not(.radio-disabled), ion-checkbox:not(.checkbox-disabled)'
-      ) as HTMLIonRadioElement | HTMLIonCheckboxElement | null;
-
-      if (firstEnabledOption) {
-        /**
-         * Focus the option for the same reason as we do above.
-         *
-         * Needs to be called before `focusVisibleElement` to prevent issue with focus event bubbling
-         * and removing `ion-focused` style
-         */
-        firstEnabledOption.setFocus();
+    };
 
-        focusVisibleElement(firstEnabledOption.closest('ion-item')!);
+    // For modals and popovers, we can scroll before they're visible
+    if (this.interface === 'modal') {
+      overlay.addEventListener('ionModalWillPresent', scrollSelectedIntoView, { once: true });
+    } else if (this.interface === 'popover') {
+      overlay.addEventListener('ionPopoverWillPresent', scrollSelectedIntoView, { once: true });
+    } else {
+      // For alerts and action sheets, we need to wait a frame after willPresent
+      const scrollAfterRender = () => {
+        requestAnimationFrame(() => {
+          scrollSelectedIntoView();
+        });
+      };
+      if (this.interface === 'alert') {
+        overlay.addEventListener('ionAlertWillPresent', scrollAfterRender, { once: true });
+      } else if (this.interface === 'action-sheet') {
+        overlay.addEventListener('ionActionSheetWillPresent', scrollAfterRender, { once: true });
       }
     }
 
+    overlay.onDidDismiss().then(() => {
+      this.overlay = undefined;
+      this.isExpanded = false;
+      this.ionDismiss.emit();
```

Comment:
Shouldn't we remove the listeners added above in some place?

Reply:
No, because they will be removed after their first execution due to [once](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#once) being set to `true`. Additionally, we clean up the `overlay` reference in the `onDidDismiss` handler.